waspc/src/Wasp/Lib.hs

```diff
@@ -173,3 +180,16 @@ compileAndRenderDockerfile waspDir compileOptions = do
     Right appSpec -> do
       dockerfileOrGeneratorErrors <- DockerGenerator.compileAndRenderDockerfile appSpec
       return $ left (map show . toList) dockerfileOrGeneratorErrors
+
+deploy :: FilePath -> Path' Abs (Dir WaspProjectDir) -> [String] -> IO ()
+deploy waspExe waspDir cmdArgs = do
+  waspDataDir <- Data.getAbsDataDirPath
+  let deployDir = waspDataDir </> [reldir|packages/deploy|]
+  unlessM (doesDirectoryExist . toFilePath $ deployDir </> [reldir|node_modules|]) $
+    runCommandAndPrintOutput $ runNodeCommandAsJob deployDir "npm" ["install"] J.Server
```

Comment:
You provided `J.Server`, but that is not really the correct context, is it? Correct context should be `J.DeployScript` or `J.Deploy` or something like that, right? What about it, how do you feel about this?

Reply:
Yeah true. I thought about adding a new one but that felt like a pretty big change. If we think the context here matters though, we can add a new one. It will just touch a lot of things. Can do just lemme know your thoughts. Could be called something less specific too, like `J.External` if we wanted to add it and be able to use in other similar contexts.

Reply:
Yeah that is also an option, so smth like J.Other or J.External or something.

If you feel it is a bit of a too big change right now, which I agree that it might be, what I would suggest is putting a TODO here explaining why you used J.Server and what we should ideally do.

Reply:
Punted with a TODO for now.